### PR TITLE
Use a configurable header for group membership

### LINKED ISSUE #268: Unable to go to the dashboard on stage
Stage has the latest code deployed, when you click "Start here" you will be redirected back and get a message:  "You are not authorized to perform the requested action".   I've added logging and determined that the application_controller never gets any value from `request.env['eduPersonEntitlement']`  This might be something that needs configuring in puppet.

Maybe we need to set `ShibUseHeaders On`?

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?
Fixes #268

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -26,9 +26,10 @@ def current_user
 
   sig { returns(T::Array[String]) }
   def ldap_groups
-    return request.env['eduPersonEntitlement'].split(';') if request.env['eduPersonEntitlement']
-
-    []
+    raw_header = request.env[Settings.authorization_group_header]
+    raw_header = ENV['ROLES'] if Rails.env.development?
+    logger.info("Roles are #{raw_header}")
```
Comment (mjgiarlo) [inline app/controllers/application_controller.rb:31]: Maybe `logger.debug` but nbd?

File: app/policies/dashboard_policy.rb [hunk 2]
```diff
@@ -5,6 +5,8 @@
 class DashboardPolicy < ApplicationPolicy
   sig { returns(T::Boolean) }
   def show?
-    user.application_user?
+    administrator? || collection_creator? # TODO: || has created any deposits
   end
+
+  delegate :administrator?, :collection_creator?, to: :user
```
Comment (mjgiarlo) [inline app/policies/dashboard_policy.rb:11]: If you're delegating here, why not also in the collection policy?

File: config/settings.yml [hunk 3]
```diff
@@ -20,7 +20,8 @@ h2:
 # feature flag
 allow_sdr_content_changes: true
 
+authorization_group_header: HTTP_X_GROUPS
 # mappings from authorization role to workgroup name
 authorization_workgroup_names:
-  administrators: 'dlss:hydrus-app-adminstrators'
+  administrators: 'dlss:hydrus-app-administrators'
```
Comment (mjgiarlo) [inline config/settings.yml:26]: :1st_place_medal:

File: app/policies/dashboard_policy.rb (hunk 2 above)
Comment (jcoyne) [reply app/policies/dashboard_policy.rb:11]: changed to be consistent.

**Review by mjgiarlo — APPROVED** [review]

File: config/settings.yml (hunk 3 above)
Comment (jmartin-sul) [reply config/settings.yml:26]: whoops!

**Pull request merged**